test/UnitTests/TestFramework.UnitTests/Assertions/AssertTests.AreEqualTests.cs

```diff
@@ -1140,4 +1140,122 @@ public bool Equals(A? other)
         public override int GetHashCode()
             => Id.GetHashCode() + 1234;
     }
+
+    // Tests for enhanced string comparison messages
+    public void AreEqualStringDifferenceShouldShowIndex()
+    {
+        Exception ex = VerifyThrows(() => Assert.AreEqual("aaaa", "aaab"));
+        Verify(ex.Message == """
+            String lengths are both 4. Strings differ at index 3.
+            Expected: "aaaa"
+            But was:  "aaab"
+            ---^ 
```

Comment:
---^ is not pointing to the right part. You need to take into consideration the biggest length between the localized text `Expected: ` and `But was: `. To do so, you will need to split the localization into multiple sub strings.